### PR TITLE
feat[SRVKP-5511]: run golangci-lint using CI image

### PR DESCRIPTION
**Pull request by srivickynesh**

### CODE REVIEW

File: .tekton/go-lint-pipelinerun.yaml [hunk 1]
```diff
@@ -38,19 +38,21 @@ spec:
...
           - fetch-repository
-        params:
-          - name: package
-            value: .
-          - name: version
-            value: "v1.55"
+        taskSpec:
+          steps:
+            - name: lint
+              image: quay.io/openshift-pipeline/ci
```
Comment (srivickynesh) [inline .tekton/go-lint-pipelinerun.yaml:46]: use this image quay.io/srivickynesh/pipelines-ci

File: Dockerfile.CI [hunk 2]
```diff
@@ -60,3 +60,6 @@ RUN wget https://github.com/sigstore/cosign/releases/download/v2.2.4/cosign-linu
 
 RUN wget https://github.com/sigstore/rekor/releases/download/v1.3.6/rekor-cli-linux-amd64 -O /usr/bin/rekor-cli && \
     chmod u+x /usr/bin/rekor-cli
+
+ENV GOLANGCI_LINT_VERSION=1.59.0
+RUN curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /usr/bin v${GOLANGCI_LINT_VERSION}
```
Comment (ppitonak) [inline Dockerfile.CI:65]: I would prefer downloading tarball from https://github.com/golangci/golangci-lint/releases/download/v1.59.1/golangci-lint-1.59.1-linux-amd64.tar.gz and extracting it. You never know what installation script is doing it's safer to keep it simple. We do the same for all other tools in this image.

**Review by ppitonak — CHANGES_REQUESTED** [review]

File: Dockerfile.CI (hunk 2 above)
Comment (srivickynesh) [reply Dockerfile.CI:65]: I tried extracting the tarball there was some issue with that executable file.

Comment (srivickynesh) [reply Dockerfile.CI:65]: /usr/bin/golangci-lint: cannot execute binary file: Exec format error

Comment (srivickynesh) [reply Dockerfile.CI]: Done

File: .tekton/go-lint-pipelinerun.yaml [hunk 3]
```diff
@@ -38,19 +38,20 @@ spec:
...
           - fetch-repository
-        params:
-          - name: package
-            value: .
-          - name: version
-            value: "v1.59"
+        taskSpec:
+          steps:
+            - name: lint
+              image: quay.io/srivickynesh/pipelines-ci #quay.io/openshift-pipeline/ci
```
Comment (ppitonak) [inline .tekton/go-lint-pipelinerun.yaml:46]: reminder to myself to check it before merging :)

**Review by ppitonak — APPROVED** [review]

**Comment by ppitonak** [comment]
@srivickynesh It looks good, please update the image and I'll merge it.

**Pull request merged**